SeafShare/SeafShareFileViewController.m

```diff
@@ -53,18 +50,16 @@ - (void)viewDidLoad {
     }
     
     self.navigationItem.title = NSLocalizedString(@"Save to Seafile", @"Seafile");
-    self.destinationLabel.text = NSLocalizedString(@"Destination", @"Seafile");
-    self.destinationButton.enabled = false;
-    [self.saveButton setTitle:NSLocalizedString(@"Save", @"Seafile") forState:UIControlStateNormal];
+    [self.cancelButton setTitle:NSLocalizedString(@"Cancel", @"Seafile") forState:UIControlStateNormal];
     
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(selectDirNotification:) name:@"SelectedDirectoryNotif" object:nil];
     [self loadInputs];
     [self setupTableview];
+    [self updateDestinationLabel];
 }
 
-- (void)viewWillAppear:(BOOL)animated {
-    [super viewWillAppear:animated];
-    [self updateSaveButton];
+- (void)viewDidAppear:(BOOL)animated {
```

Comment:
Do not call startUpload in viewDidAppear. Because viewDidAppear might be called multiple times, e.g. user go to home screen and then go back. If you really want to do it inside viewDidAppear, make sure startUpload will be called only once or make sure it is idempotent